firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java

```diff
@@ -147,117 +169,51 @@ public void addToCollectionParentIndex(ResourcePath collectionPath) {
   public void addFieldIndex(FieldIndex index) {
     hardAssert(started, "IndexManager not started");
 
-    int nextIndexId = memoizedMaxId + 1;
+    int nextIndexId = memoizedMaxIndexId + 1;
```

Comment:
I am confused about how we initialize memoizedIndexId. It's currently initialized to 1, but where do we read the current max value from the table? FWIU, we're setting new field indexes to indexid 0 across restarts.

I have the same question for memoizedMaxSequenceNumber. I'm not sure where we're reading the initial value from persistence.

Reply:
Update, realized we take care of this in memoizeIndex(). Could you document that somewhere? The call to `memoizeIndex` is hidden  deep inside `start()`, and it'd be helpful to clarify that the memoized max counter values are only calculated when we memoize.

However, this also creates a potential problem. What happens if a user's max index id or seq number is not the global max for the table? won't we start overriding the existing counter values? I think it is safer to check for the memoized max ids as separate SELECT MAX calls from the table during start().

Reply:
Not sure if it is within scope for this PR, but we need a TODO somewhere to write tests for user changes.